tasks/collector.py

```diff
@@ -35,7 +58,70 @@
     },
 }
 
-
+class YAMLValidationError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+def validate_manifest(manifest): # return list of components to remove, empty list if valid
+    # validate collector version matches ocb version
+    version_checks = ["version", "otelcol_version"]
+    for version in version_checks:
+        manifest_version = manifest.get("dist").get(version)
+        if manifest_version != OCB_VERSION:
+            raise YAMLValidationError(f"Collector version ({manifest_version}) in manifest does not match required OCB version ({OCB_VERSION})")
```

Comment:
This won't work if the user is not setting this attribute explicitly. Do we want to require this attribute?

Reply:
so to my understanding, we are requiring otelcol_version to be set to a specific version, but it sounds like version is just for your own collector so we could remove that one.

Reply:
We are requiring you to have `otelcol_version` set to a specific value. What I am saying is that it would also be fine if it's not set to any value at all, since the default will work

Reply:
ah, ok thanks. I'll check to see if they set the otelcol_version, and if so, return an error if it doesn't match OCB_VERSION or continue otherwise. I'll drop the "version" check since I think that was just me misunderstanding the assignment

Reply:
fixed; only checking otelcol_version now and if that value is empty, don't throw an error. added extra test cases as well.